src/integer_mod_q/polynomial_ring_zq/coefficient_embedding.rs

```diff
@@ -0,0 +1,236 @@
+// Copyright © 2025 Marcel Luca Schmidt
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains implementations to transform a [`PolynomialRingZq`]
+//! into a [`MatZq`] and a [`ModulusPolynomialRingZq`] and reverse by using the coefficient embedding.
+
+use crate::{
+    integer::Z,
+    integer_mod_q::{MatZq, ModulusPolynomialRingZq, PolynomialRingZq},
+    traits::{
+        FromCoefficientEmbedding, GetCoefficient, GetEntry, GetNumRows, IntoCoefficientEmbedding,
+        SetCoefficient, SetEntry,
+    },
+};
+
+impl IntoCoefficientEmbedding<(MatZq, ModulusPolynomialRingZq)> for &PolynomialRingZq {
+    /// Computes the coefficient embedding of the polynomial
+    /// in a [`MatZq`] as a column vector, where the i-th entry
+    /// of the vector corresponds to the i-th coefficient, and a
+    /// [`ModulusPolynomialRingZq`].
+    /// It inverts the operation of [`PolynomialRingZq::from_coefficient_embedding`].
+    ///
+    /// The representation of the polynomials in the embedding is in the range `[0, modulus_polynomial)`.
+    ///
+    /// Parameters:
+    /// - `size`: determines the number of rows of the embedding. It has to be larger
+    ///     than the degree of the polynomial.
+    ///
+    /// Returns a coefficient embedding as a column vector if `size` is large enough.
+    ///
+    /// # Examples
+    /// ```
+    /// use std::str::FromStr;
+    /// use qfall_math::{
+    ///     integer_mod_q::{MatZq, PolynomialRingZq},
+    ///     traits::IntoCoefficientEmbedding,
+    /// };
+    ///
+    /// let poly = PolynomialRingZq::from_str("2  1 -2 / 3  17 3 5 mod 19").unwrap();
+    /// let embedding = poly.into_coefficient_embedding(3);
+    /// let cmp_vector = MatZq::from_str("[[1],[-2],[0]] mod 19").unwrap();
+    /// assert_eq!((cmp_vector, poly.get_mod()), embedding);
+    /// ```
+    ///
+    /// # Panics ...
+    /// - if `size` is not larger than the degree of the polynomial, i.e.
+    ///     not all coefficients can be embedded.
+    fn into_coefficient_embedding(self, size: impl Into<i64>) -> (MatZq, ModulusPolynomialRingZq) {
+        let size = size.into();
+        let length = self.get_degree() + 1;
+        assert!(
+            size >= length,
+            "The polynomial can not be embedded in the vector, \
+            as the length of the polynomial ({length}) is larger than \
+            the provided size ({size})."
+        );
+        let mut out = MatZq::new(size, 1, self.modulus.get_q());
+        for j in 0..size {
+            let coeff: Result<Z, _> = self.get_coeff(j);
+            match coeff {
+                Ok(value) => out.set_entry(j, 0, value).unwrap(),
+                Err(_) => break,
+            }
+        }
```

Comment:
```suggestion
        for j in 0..self.get_degree() {
            let coeff: Z = self.get_coeff(j).unwrap();
            out.set_entry(j, 0, value).unwrap();
        }
```
If you iterate over the size of the polynomial, you should be guaranteed to not receive any errors (and be more efficient as you leave out iterations for any 0 entries at the bottom).

Reply:
Yes, this is true. It is a copy of our underlying code for all coefficient embeddings. I will change this in all instances.